code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,1135 @@
+			return
+
+		person_fed.nutrition += 100 //the equivalent of one nutrition level
+		nutriment -= 5
+		desc = "Feed me and I will feed you back, I currently hold [nutriment] units of nutrients."
+
+/obj/effect/warped_rune/silverspace/Destroy()
+	nutriment = null
+	return ..()
+
+/* Bluespace rune,reworked so that the last person that walked on the rune will swap place with the next person stepping on it*/
+
+
+/obj/item/slimecross/warping/bluespace
+	colour = "bluespace"
+	runepath = /obj/effect/warped_rune/bluespace
+	effect_desc = "Puts up a rune that will swap the next two person that walk on the rune."
+
+
+obj/effect/warped_rune/bluespace
+	desc = "Everyone is everywhere at once, yet so far away from each other"
+	icon_state = "bluespace_rune"
+	max_cooldown = 10 //only here to avoid spam lag
+	/// first person to run into the rune
+	var/mob/living/carbon/first_person
+	///second person that run into the rune
+	var/mob/living/carbon/second_person
+	///here to remember if the rune has been stepped on before
+	var/stepped_on = 0
+
+
+///the first two person that stepped on the rune swap places after the second person stepped on it.
+obj/effect/warped_rune/bluespace/Crossed(atom/movable/crossing)
+	. = ..()
+	if(cooldown > world.time) //checks if 2 seconds have passed to avoid spam.
+		return
+
+	cooldown = max_cooldown + world.time
+	if(!istype(crossing,/mob/living/carbon/human))
+		return
+
+	if(stepped_on == 0)
+		first_person = crossing //remember who stepped in so we can teleport them later.
+		stepped_on++
+		return
+
+	if(crossing == first_person)
+		return
+
+	second_person = crossing
+	do_teleport(second_person, first_person, forceMove = TRUE)//swap both of their place.
+	do_teleport(first_person, rune_turf, forceMove = TRUE)
+	stepped_on--
+
+
+obj/effect/warped_rune/bluespace/Destroy()
+	first_person = null
+	second_person = null
+	return ..()
+
+
+/* will make anyone stepping over it older and much slower for a few second, step on it too much or be too old and you'll get dusted.*/
+
+
+/obj/item/slimecross/warping/sepia
+	colour = "sepia"
+	runepath = /obj/effect/warped_rune/sepiaspace
+	effect_desc = "Draws a rune that make people grow older and slower until they eventually wither away."
+	drawing_time = 200 //much longer to draw than other runes because it fucking stops time
+
+
+obj/effect/warped_rune/sepiaspace
+	icon_state = "time_space"
+	desc = "The clock is ticking, but in what direction?"
+
+
+///slows down whoever walks on the rune and makes them older by ten years. if the person goes above 100 years they get dusted.Very similar to the "old timer" reagent code with a few tweaks.
+obj/effect/warped_rune/sepiaspace/Crossed(atom/movable/crossing)
+	. = ..()
+	var/mob/living/carbon/human/slowed_down
+	if(istype(crossing,/mob/living/carbon/human))
+		slowed_down = crossing
+		slowed_down.add_movespeed_modifier(/datum/movespeed_modifier/status_effect/sepia_rune)
```

Comment:
if you walk across this rune, then walk back across it while you're still slow, the slowdown effect won't be renewed because of this.

but if you walk across the rune, walk back across it before the slowdown effect expires, wait for the slowdown effect to expire, then walk back across the rune, the new slowdown effect will end "early" when the slowdown effect from the second crossing "should" have ended.